### PR TITLE
Error conversions

### PR DESCRIPTION
**Pull request by jchengjr77**
Converted Canaveral's codebase to not panic upon error. Now, all errors should be handled within the function, and either ignored or returned. Returned errors (which are most if not all the errors) will bubble up to the top level (main.go) and be handled appropriately there.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -1,66 +1,118 @@
...
+      - name: Test (node)
+        run: go test ./node
+      - name: Test (vscodesupport)
+        run: go test ./vscodesupport
+      - name: Test (gh)
+        run: go test ./gh
+      - name: Test (git)
+        run: go test ./git
+      - name: Test (finder)
+        run: go test ./finder
```
Comment (SeanPrendi) [inline .github/workflows/go.yml:60]: See my comment on #17 about adding back python

File: addmanager.go [hunk 2]
```diff
@@ -72,7 +85,10 @@ func addProjectHandler(projName string, projType string, init bool) error {
 		return errors.New("Unsupported project type specified")
 	}
 	if init {
-		createAndInit(projName)
+		err := createAndInit(projName)
+		if err != nil {
+			return err
+		}
```
Comment (SeanPrendi) [inline addmanager.go:91]: Can this just be `return err`? Seems like if `err == nil` then you just end the case and return `nil` anyway.

File: main.go [hunk 3]
```diff
@@ -322,8 +318,25 @@ func main() {
...
+				Argument should be a project name.`,
+				Usage: "Opens selected project in a finder window",
+				Action: func(c *cli.Context) error {
+					if qFlag {
+						fmt.Println("(okay, I'll try to be quiet.)")
+					}
+					projName := c.Args().Get(0)
+					err := finder.OpenFinder(
+						projName, usrHome+confDir+wsFName)
+					return err
```
Comment (SeanPrendi) [inline main.go:339]: Might need to check here that `c.Args().Len() >= 1` (or `==1`) to prevent null pointer dereference if the user does `canaveral finder` with no argument.

File: addmanager.go (hunk 2 above)
Comment (jchengjr77) [reply addmanager.go:91]: Yes, that should be changed. Let me do that...

File: main.go (hunk 3 above)
Comment (jchengjr77) [reply main.go:339]: This is covered in a test case (null pointers actually get handled as empty strings, don't ask me why). I also just manually tested it to see if the null pointer is an issue, and it looks good. Thanks for highlighting this.
